### DISCUSSION TITLE
Gioco di logica in Home Assistant! Ho bisogno di un fisico :)

### DISCUSSION THREAD

**Discussion by ElyFNV** [indomus/forum#484] [GENERALE - Discussioni generiche]
Ciao a tutti!

Ho acquistato un sensore di luminosità su Amazon e un sensore di movimento Hue sempre su Amazon.

Sono riuscito ad integrarli perfettamente su Home Assistant grazie all'usb ConBee II.

Il mio obiettivo era quello di creare un'automazione che accendesse le luci nella stanza quando rilevasse dei movimenti e SOLO se la luminosità (rilevata dal sensore di luce) fosse inferiore ad un certo valore da me impostato. Ci sono però due importanti fattori da considerare:

1. Il sensore di movimento Hue ha un tempo di aggiornamento di soli 15 secondi

2. Il sensore di luminosità si aggiorna solo 10 secondi dopo una cospicua variazione di luminosità

Questo pone un grosso problema nell'esecuzione dell'automazione.

Mi spiego con un esempio pratico:

- Entro in una stanza buia. L'automazione si avvia, perché il sensore di movimento mi rileva (trigger) e perché la luminosità è inferiore al cut-off impostato (condizione).

- Nel frattempo mi trasferisco in un'altra stanza. Il sensore di movimento attende 15 secondi prima di analizzare nuovamente l'area (come menzionato sopra). Dopo 15 secondi, il sensore di movimento controlla l'area e non vede alcun movimento, quindi spegne le luci nella stanza (grazie a un'altra semplice automazione che ho creato).

- Tuttavia, trascorrono 10 secondi dal momento in cui le luci si spengono a quando il sensore di luminosità trasmette il nuovo valore a Home Assistant!!!

- In questi 10 secondi torno nella stanza, che ora è buia. Tuttavia, come già accennato, il valore di luminosità in HA non è stato ancora aggiornato, perché non sono ancora trascorsi 10 secondi. Di conseguenza, l'automazione fallisce! Perché il sensore di movimento rileva il movimento (trigger) ma il valore di luminosità presente nella stanza (condizione) è ancora quello vecchio di quando la luce era accesa e quindi risulta essere maggiore del cut-off.

E' un po' un cane che si morde la coda...

Qualcuno può aiutarmi? Come potrei risolvere questo problema?

Grazie mille a tutti in anticipo!!!

**Comment by 50494554524F** [comment]
puoi usare 

homeassistant.update_entity

per richiedere un aggiornamento del sensore... naturalmente, se i 10-15s sono codificati nel firmware non puoi farci niente...
